icu4j/main/classes/core/src/com/ibm/icu/impl/ICUResourceBundle.java

```diff
@@ -1304,17 +1388,16 @@ public ICUResourceBundle load() {
 
             // fallback to locale ID parent
             if(b == null){
-                int i = localeName.lastIndexOf('_');
-                if (i != -1) {
-                    // Chop off the last underscore and the subtag after that.
-                    String temp = localeName.substring(0, i);
-                    b = instantiateBundle(baseName, temp, defaultID, root, openType);
+                String origLocaleName = (origLocaleID != null) ? origLocaleID : localeName;
+                String fallbackLocaleID = getParentLocaleID(localeName, origLocaleName);
```

Comment:
This looks like the place where we should check the openType.
If it's LOCALE_DEFAULT_ROOT or LOCALE_ONLY then call getParentLocaleID().
Otherwise, use the old truncation; maybe wrap it into a little helper function that also returns null if there is no more underscore.

Reply:
Yup.